proto/ibc/applications/fee/v1/query.proto

```diff
@@ -0,0 +1,123 @@
+syntax = "proto3";
+
+package ibc.applications.fee.v1;
+
+import "gogoproto/gogo.proto";
+import "cosmos/base/query/v1beta1/pagination.proto";
+import "ibc/applications/fee/v1/fee.proto";
+import "google/api/annotations.proto";
+import "ibc/core/channel/v1/channel.proto";
+import "cosmos/base/query/v1beta1/pagination.proto";
+
+option go_package = "github.com/cosmos/ibc-go/modules/apps/29-fee/types";
+
+// Query provides defines the gRPC querier service.
+service Query {
+  // Gets the fee expected for submitting ReceivePacket msg for the given packet
+  rpc ReceiveFee(QueryReceiveFeeRequest) returns (QueryReceiveFeeResponse) {
+    option (google.api.http).get =
+        "/ibc/apps/fee/v1/receive_fee/port/{packet_id.port_id}/channel/{packet_id.channel_id}/sequence/"
+        "{packet_id.sequence}/relayer/{relayer_address}/height/{query_height}";
+  }
+
+  // Gets the fee expected for submitting AcknowledgePacket msg for the given packet
+  rpc AckFee(QueryAckFeeRequest) returns (QueryAckFeeResponse) {
+    option (google.api.http).get =
+        "/ibc/apps/fee/v1/ack_fee/port/{packet_id.port_id}/channel/{packet_id.channel_id}/sequence/"
+        "{packet_id.sequence}/relayer/{relayer_address}/height/{query_height}";
+  }
+
+  // Gets the fee expected for submitting TimeoutPacket msg for the given packet
+  rpc TimeoutFee(QueryTimeoutFeeRequest) returns (QueryTimeoutFeeResponse) {
+    option (google.api.http).get =
+        "/ibc/apps/fee/v1/timeout_fee/port/{packet_id.port_id}/channel/{packet_id.channel_id}/sequence/"
+        "{packet_id.sequence}/{relayer_address}/height/{query_height}";
+  }
+
+  // Gets all incentivized packets
+  rpc IncentivizedPackets(QueryIncentivizedPacketsRequest) returns (QueryIncentivizedPacketsResponse) {
+    option (google.api.http).get = "/ibc/apps/fee/v1/incentivized_packets/height/{query_height}";
+  }
+
+  // Gets all incentivized packets
+  rpc IncentivizedPacket(QueryIncentivizedPacketRequest) returns (QueryIncentivizedPacketResponse) {
+    option (google.api.http).get =
+        "/ibc/apps/fee/v1/incentivized_packet/port/{packet_id.port_id}/channel/{packet_id.channel_id}/sequence/"
+        "{packet_id.sequence}/height/{query_height}";
+  }
+}
+
+// QueryReceiveRequestFee is the request type for quering the receive fee
+message QueryReceiveFeeRequest {
+  // PacketID
+  ibc.core.channel.v1.PacketId packet_id = 1 [(gogoproto.moretags) = "yaml:\"packet_id\""];
```

Comment:
sorry for mentioning this so late, but `query.proto` doesn't need yaml for some reason. At least [SDK](https://github.com/cosmos/cosmos-sdk/blob/bf049d36d1f955302f0f4aacadb773a90496813c/proto/cosmos/group/v1beta1/query.proto#L130) doesn't do it (neither do [we](https://github.com/cosmos/ibc-go/blob/0afa15af41997cfb4c5dd5f98de8043961e26195/proto/ibc/core/client/v1/query.proto#L64)). I will remove them